vertical-pod-autoscaler/enhancements/4016-in-place-updates-support/README.md

```diff
@@ -17,88 +17,133 @@
-* `InPlaceOrRecreate`.
+* `InPlaceOrRecreate`
+
+Here we specify `OrRecreate` to make sure the user explicitly knows that the container may be
+restarted.
 
 [`UpdateMode`]: https://github.com/kubernetes/autoscaler/blob/71b489f5aec3899157b37472cdf36a1de223d011/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1/types.go#L124
 
 ## Context
 
-[In-place update of pod resources KEP] is available in alpha in [Kubernetes 1.27]. The feature allows changing container
-resources while the container is running. It also adds [`ResizePolicy`] field to Container. This field indicates for an
-individual resource if a container needs to be restarted by kubelet when the resource is changed. For example it may be
-the case that a Container automatically adapts to a change in CPU, but needs to be restarted for a change in Memory to
-take effect.
+[In-place update of pod resources KEP] is available in alpha in 1.27 and graduating to beta in
+1.33. The feature allows changing container resources while the container is running. It adds
+two key features:
+
+* A [`/resize` subresource] that can be used to mutate the `Pod.Spec.Containers[i].Resources`
+  field.
+* A [`ResizePolicy`] field to Container. This field allows to the user to specify the behavior when
+  modifying a resource value. Currently it has two modes:
+  - `NotRequired` (default) which indicates to the container runtime that it should try to resize
+    the container without restarting. However, it does not guarantee that a restart will not
+    happen.
+  - `RestartContainer` which indicates that any mutation to the resource requires a restart (for
+    example, this is important for Java apps using the `-xmxN` which are unable to resize memory
+    without restarting).
+
+Note that resize operations will NOT change the pod's quality of service (QoS) class.
 
 [In-place update of pod resources KEP]: https://github.com/kubernetes/enhancements/issues/1287
-[Kubernetes 1.27]: https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.27.md#api-change-3
-[`ResizePolicy`]: https://github.com/kubernetes/api/blob/8360d82aecbc72aa039281a394ebed2eaf0c0ccc/core/v1/types.go#L2448
+[`/resize` subresource]:https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1287-in-place-update-pod-resources#api-changes
+[`ResizePolicy`]: https://github.com/kubernetes/api/blob/4dccc5e86b957cea946a63c4f052ee7dec3946ce/core/v1/types.go#L2636
 
 ## Design Details
 
-In the update modes existing before (`Initial` and `Recreate`) only admission controller was changing pod spec (updater
-was responsible for evicting pods so that admission controller could change them during admission).
+Today, only the VPA admission controller is responsible for changing the pod spec.
+
+The VPA updater is responsible for evicting pods so that the admission controller can change them
+during admission.
+
+In the newly added `InPlaceOrRecreate` mode, the VPA Updater will attempt to execute in-plac
```

Comment:
I saw maxcao13@'s comments and I also feel that it's not completely clear what the relationship between the new VPA mode and the container restart resize policies is. It's probably worth it to document it here. 

For example:
For `PreferNoRestart` resize restart policy, VPA in `InPlaceOrRecreate` will attempt to execute in-place updates FIRST. If it is unable to process in-place updates in conditions X, Y, Z, it will restart/evict the pod to force a change. 

For `RestartContainer`, VPA in `InPlaceOrRecreate` will attempt to restart the container first. Or recreate the pod if the resize fails.

Reply:
Documented this in a new section, PTAL.